src/nvme/nbft.c

```diff
@@ -0,0 +1,739 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/*
+ * This file is part of libnvme.
+ * Copyright (c) 2021-2022, Dell Inc. or its subsidiaries.  All Rights Reserved.
+ *
+ * Authors: Stuart Hayes <Stuart_Hayes@Dell.com>
+ *
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <arpa/inet.h>
+#include <ccan/endian/endian.h>
+
+#include "private.h"
+#include "nbft.h"
+#include "log.h"
+
+
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+static __u8 csum(void *buffer, int length)
```

Comment:
buffer should propebly be `const __u8 *` type, this avoids the cast below and the caller is actually passing in `__u8 *`

length should be type `ssize_t` because of the `raw_nbft_size` definition.

Reply:
fixed with my latest update.